tests/unit/data/test_folder.py

```diff
@@ -0,0 +1,47 @@
+"""Unit Tests - Folder Datamodule."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from anomalib.data import Folder, TaskType
+from .base import _TestAnomalibImageDatamodule
+import pytest
+
+
+class TestFolder(_TestAnomalibImageDatamodule):
+    """Folder Datamodule Unit Tests.
+
+    All of the folder datamodule tests are placed in ``TestFolder`` class.
+    """
+
+    @pytest.fixture(params=[None, "good_test"])
+    def normal_test_dir(self, request) -> str:
+        """Create and return a normal test directory."""
+        return request.param
+
+    @pytest.fixture
+    def datamodule(self, task_type: TaskType, normal_test_dir: str) -> Folder:
+        """Create and return a folder datamodule.
+
+        This datamodule uses the MVTec bottle dataset for testing.
+        """
+        # Make sure to use a mask directory for segmentation. Folder datamodule
+        # expects a relative directory to the root.
+        mask_dir = None if task_type == TaskType.CLASSIFICATION else "ground_truth/broken_large"
+
+        # Create and prepare the dataset
+        _datamodule = Folder(
+            root="./datasets/bottle",
```

Comment:
We should probably use `get_dataset_path` for all the dataset tests. It would be a good idea in the future to make this a fixture instead of a method.